core/conversion/converters/impl/element_wise.cpp

```diff
@@ -326,8 +326,24 @@ auto element_wise_registrations TORCHTRT_UNUSED =
                } else if (rounding_mode == "trunc") {
                  // trunc = floor(abs(div)) * sign(div)
                  auto tmp_div = add_elementwise(ctx, nvinfer1::ElementWiseOperation::kDIV, self, other, "tmp_div");
-                 auto abs = ctx->net->addUnary(*tmp_div->getOutput(0), nvinfer1::UnaryOperation::kABS);
-                 auto floor = ctx->net->addUnary(*abs->getOutput(0), nvinfer1::UnaryOperation::kFLOOR);
+                 auto abs = add_absolute_value(ctx, n, tmp_div->getOutput(0), util::node_info(n) + "_absolute_val");
+
+                 // In this case, we allow the floor unary on non-TRT Unary types, as it is needed for this
+                 // specific function. Floor applied to non-float types equates to identity
+                 nvinfer1::ILayer* floor;
+                 if ((abs->getOutput(0)->getType() == nvinfer1::DataType::kINT32) ||
+                     (abs->getOutput(0)->getType() == nvinfer1::DataType::kBOOL)) {
+                   LOG_GRAPH(
+                       "Tensor is of unsupported type " << abs->getOutput(0)->getType()
+                                                        << " for IUnaryLayer::kFLOOR. Using identity instead.");
+                   floor = ctx->net->addIdentity(*abs->getOutput(0));
+                   TORCHTRT_CHECK(floor, "Unable to create identity layer from node: " << *n);
+                 } else {
+                   floor = ctx->net->addUnary(*abs->getOutput(0), nvinfer1::UnaryOperation::kFLOOR);
+                   TORCHTRT_CHECK(floor, "Unable to create floor layer from node: " << *n);
+                 }
+                 floor->setName((util::node_info(n) + "_floor").c_str());
```

Comment:
In this code block, both the `abs` and the `floor` operators were encountering errors when both inputs are integer types. The solution for `abs` was a converter utility, whereas for `floor`, the solution only appears here. The reasoning for this choice was that Torch support for `torch.floor()` applied to an Int32 type differs between `1.13.0.dev20220921+cu116` and `1.14.0.dev20221018+cu116`, so it is unclear if the `aten::floor` converter should generally support Int32 inputs or not, currently.